### ISSUE TITLE
how to add val dataset' MAP?

### ISSUE THREAD

**Issue by dubiwei** [sfzhang15/RefineDet#122]
Hello:
       Do you know how to add the val dataset' MAP when training? I just started learning caffe.
Thanks

**Comment by sfzhang15** [comment]
@dubiwei 
You can refer to [this example](https://github.com/BVLC/caffe/blob/master/examples/mnist/lenet_auto_solver.prototxt) from CAFFE.

**Comment by dubiwei** [comment]
thanks, I add the val dataset, but the MAP of val dataset is much larger than the test dataset.
Does the val dataset take part in training?

**Comment by sfzhang15** [comment]
No, the val subset does not take part in training.

**Comment by dubiwei** [comment]
and if I want to resize the input image to another size , what should I do ? thanks

**Comment by sfzhang15** [comment]
@dubiwei 
1. RefineDet dose not have the region-wise operations (i.e., RoIPooling).
2. If you use the python interface of caffe, you can use resize function before inputing.

**Comment by dubiwei** [comment]
The input image size must be 512 or 320? if i use another size,what should i do?thanks

**Comment by sfzhang15** [comment]
The input image size can be any size if your GPU has enough memory. You can change the resize_width and resize_height.

**Comment by dubiwei** [comment]
but if only set the resize shape, it will raise the error:bottom[i]->shape()==bottom[0]->shape()

**Comment by sfzhang15** [comment]
@dubiwei 
Yes, since the RefineDet has the deconvolution layer and it only supports the integer multiple scaling. So the image size should be the multiple of 64.

**Comment by sfzhang15** [comment]
@dubiwei 
If you want to training the model with any size, you can replace the deconvolution layer with the upsampling layer implemented by the third-party.
